_posts/2019-06-07-python-datascience-with-openfaas.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Serving Data Science Models with OpenFaaS
+# TODO
+description: "Build and deploy a name classification function with OpenFaaS [<--FUN SUGGESTIONS WELCOME]"
+date: 2019-06-01
+image: /images/multifile-python-function/pexels-book-stack-books-bookshop-264635.jpg
+categories:
+  - python
+  - coding
+  - examples
+  - datascience
+  - templates
+author_staff_member: lucas
+dark_background: true
+---
+
+Last month OpenFaaS was at [Kubecon Barcelona][kubecon-homepage], a frequent topic was deploying data science models to Kubernetes and, of course, can OpenFaaS help deploy models? In this post we will introduce a new function template aimed at Python data scientists and walk through a concrete example of deploying a [PyTorch][pytorch-homepage] model.
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Up bright and early? We&#39;re meeting for an OpenFaaS f2f with end-users, contributors and community. 10.20am outside the keynote hall. See you there? <a href="https://twitter.com/hashtag/serverless?src=hash&amp;ref_src=twsrc%5Etfw">#serverless</a> <a href="https://twitter.com/hashtag/community?src=hash&amp;ref_src=twsrc%5Etfw">#community</a> <a href="https://twitter.com/hashtag/faas?src=hash&amp;ref_src=twsrc%5Etfw">#faas</a> <a href="https://twitter.com/hashtag/microservices?src=hash&amp;ref_src=twsrc%5Etfw">#microservices</a> <a href="https://twitter.com/hashtag/gitops?src=hash&amp;ref_src=twsrc%5Etfw">#gitops</a> <a href="https://t.co/Af3BmXvaKl">pic.twitter.com/Af3BmXvaKl</a></p>&mdash; OpenFaaS (@openfaas) <a href="https://twitter.com/openfaas/status/1131450940807098368?ref_src=twsrc%5Etfw">May 23, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+## The `pydatascience template`
+The [pydatascience template][pydata-template] showcases a couple of ideas that anyone can leverage in their own functions and templates:
+
+1. using the [Conda package manager][conda-homepage]
+2. setting up a non-root user Python environment
+3. [multi-module Python function][multifile-blog-post]
+4. using HTTP mode in [of-watchdog][of-watchdog-homepage] to load an asset (in this case a model) into memory during startup
+
+## The name classifier function
+
+In this post, we share an example [name classifier][name-classifier-repo].  It is a relatively simple function that accepts a `name` and then attempts to guess the nationality of that name, returning the top three guesses.
+
+```sh
+echo "Lucas" | faas-cli invoke classify
+[
+  ["(-1.4300)","Irish"],
+  ["(-1.6112)","Italian"],
+  ["(-2.0314)","Dutch"]
+]
+```
+
+You can try this in your own OpenFaaS cluster using
+
+```sh
+faas-cli deploy --image=theaxer/classify:latest --name=classify
+```
+
+Caution, I did not say it would be a great guesser. This model was built on a relatively small data set and is intended to be a simple demonstration. And remember, with all things data science, "with great power comes great responsibility".
```

Comment:
`Caution, I did not say it would be a great guesser.`

Please remove or make more objective / scientific - by "not great" = "not accurate" / talk about the % accuracy you have measured.

Reply:
i will just remove it